tests/test_request.py

```diff
@@ -520,3 +543,51 @@ def protocol(app, event_loop):
     resp = await client.post('/test', data='{"foo')
     assert resp.status == HTTPStatus.BAD_REQUEST
     assert resp.body == b'Unparsable JSON body'
+
+
+async def test_cannot_consume_lazy_body(client, app):
+
+    @app.route('/test', methods=['POST'], lazy=True)
+    async def post(req, resp):
+        with pytest.raises(HttpError):
+            resp.body = req.body
+        resp.body = "Error has been raised"
+
+    resp = await client.post('/test', data='blah')
+    assert resp.status == HTTPStatus.OK
+    assert resp.body == b'Error has been raised'
+
+
+async def test_can_consume_lazy_body_if_manually_loaded(client, app):
+
+    @app.route('/test', methods=['POST'], lazy=True)
+    async def post(req, resp):
+        await req.read()
+        resp.body = req.body
+
+    resp = await client.post('/test', data='blah')
+    assert resp.status == HTTPStatus.OK
+    assert resp.body == b'blah'
+
+
+async def test_can_consume_lazy_request_using_read(client, app):
+
+    @app.route('/test', methods=['POST'], lazy=True)
+    async def post(req, resp):
+        resp.body = await req.read()
+
+    resp = await client.post('/test', data='blah')
+    assert resp.status == HTTPStatus.OK
+    assert resp.body == b'blah'
+
+
+async def test_can_consume_lazy_request_iterating(client, app):
+
+    @app.route('/test', methods=['POST'], lazy=True)
+    async def post(req, resp):
+        async for chunk in req:
+            resp.body = chunk
+
+    resp = await client.post('/test', data='blah'*1000)
+    assert resp.status == HTTPStatus.OK
+    assert resp.body == b'blah'*1000
```

Comment:
Ah, liveclient tests seem to be missing ?

Reply:
I don't get why that test is not passing anymore, I need to dig into that.

Reply:
I pushed the test (not passing).

Reply:
Wait, it is passing on the CI 🤔

Reply:
It doesn't look to pass under python 3.6 with default macOS python, updating to breaw's python3.7 made the test passing.